src/Dev/TestSession.php

```diff
@@ -202,44 +201,55 @@ public function sendRequest($method, $url, $data, $headers = null, $session = nu
      * Wrong: array('mycheckboxvalues' => array(1 => 'one', 2 => 'two'))
      * Right: array('mycheckboxvalues[1]' => 'one', 'mycheckboxvalues[2]' => 'two')
      *
-     * @see http://www.simpletest.org/en/form_testing_documentation.html
-     *
-     * @param string $formID HTML 'id' attribute of a form (loaded through a previous response)
-     * @param string $button HTML 'name' attribute of the button (NOT the 'id' attribute)
-     * @param array $data Map of GET/POST data.
-     * @return HTTPResponse
+     * $formID HTML 'id' attribute of a form (loaded through a previous response)
+     * $button HTML 'name' attribute of the button (NOT the 'id' attribute)
+     * $data Map of GET/POST data.
+     * $withSecurityToken Submit with the form's security token if there is one.
      * @throws Exception
      */
-    public function submitForm($formID, $button = null, $data = [])
+    public function submitForm(string $formID, string $button = null, array $data = [], bool $withSecurityToken = true): HTTPResponse
     {
+        /** @var Crawler $page */
         $page = $this->lastPage();
         if ($page) {
-            $form = $page->getFormById($formID);
-            if (!$form) {
-                user_error("TestSession::submitForm failed to find the form {$formID}");
+            try {
+                $formCrawler = $page->filterXPath("//form[@id='$formID']");
+                $form = $formCrawler->form();
+            } catch (InvalidArgumentException $e) {
+                return (new HTTPResponse("TestSession::submitForm failed to find the form {$formID}", 404));
```

Comment:
An error is an error and should fail the test - returning a response might give false positive results.